base_layer/wallet_ffi/src/lib.rs

```diff
@@ -1365,6 +1422,159 @@ pub unsafe extern "C" fn completed_transaction_get_destination_public_key(
+/// ptr::null_mut() if transaction is null, if the transaction status is Pending, or if the number of kernels is not
+/// exactly one.
+///
+/// # Safety
+/// The ```excess_destroy``` method must be called when finished with a TariExcess to prevent a memory leak
+#[no_mangle]
+pub unsafe extern "C" fn completed_transaction_get_excess(
+    transaction: *mut TariCompletedTransaction,
+    error_out: *mut c_int,
+) -> *mut TariExcess
+{
+    let mut error = 0;
+    ptr::swap(error_out, &mut error as *mut c_int);
+    if transaction.is_null() {
+        error = LibWalletError::from(InterfaceError::NullError("transaction".to_string())).code;
+        ptr::swap(error_out, &mut error as *mut c_int);
+        return ptr::null_mut();
+    }
+
+    // check the tx is not in pending state
+    if matches!((*transaction).status, TransactionStatus::Pending) {
+        let msg = "That transaction is still in Pending state.".to_string();
+        error = LibWalletError::from(TransactionError::StatusError(msg)).code;
+        ptr::swap(error_out, &mut error as *mut c_int);
+        return ptr::null_mut();
+    }
+
+    let kernels = (*transaction).transaction.get_body().kernels();
+
+    // currently we presume that each CompletedTransaction only has 1 kernel
+    // if that changes this will need to be accounted for
+    if kernels.len() != 1 {
+        let msg = format!("Expected 1 kernel, got {}", kernels.len());
+        error = LibWalletError::from(TransactionError::KernelError(msg)).code;
+        ptr::swap(error_out, &mut error as *mut c_int);
+        return ptr::null_mut();
+    }
+
+    let x = kernels[0].excess.clone();
+    Box::into_raw(Box::new(x))
+}
+
+/// Gets the TariExcessPublicNonce of a TariCompletedTransaction
+///
+/// ## Arguments
+/// `transaction` - The pointer to a TariCompletedTransaction
+/// `error_out` - Pointer to an int which will be modified to an error code should one occur, may not be null. Functions
+/// as an out parameter.
+///
+/// ## Returns
+/// `*mut TariExcessPublicNonce` - Returns the transaction excess public nonce, note that it will be
+/// ptr::null_mut() if transaction is null, if the transaction status is Pending, or if the number of kernels is not
+/// exactly one.
+///
+/// # Safety
+/// The ```nonce_destroy``` method must be called when finished with a TariExcessPublicNonce to prevent a memory leak
+#[no_mangle]
+pub unsafe extern "C" fn completed_transaction_get_public_nonce(
+    transaction: *mut TariCompletedTransaction,
+    error_out: *mut c_int,
+) -> *mut TariExcessPublicNonce
+{
+    let mut error = 0;
+    ptr::swap(error_out, &mut error as *mut c_int);
+    if transaction.is_null() {
+        error = LibWalletError::from(InterfaceError::NullError("transaction".to_string())).code;
+        ptr::swap(error_out, &mut error as *mut c_int);
+        return ptr::null_mut();
+    }
+
+    // check the tx is not in pending state
+    if matches!((*transaction).status, TransactionStatus::Pending) {
```

Comment:
Check for imported status as discussed above

Reply:
will fix